### PR TITLE
[CodeCompletion] Annotated result type name

### PR DESCRIPTION
**Pull request by rintaro**
Introduced 'TypeAnnotationBegin' chunk kind for grouping the result type name chunks.

rdar://problem/62617558

### CODE REVIEW

**Comment by rintaro** [comment]
@swift-ci Please smoke test

**Comment by rintaro** [comment]
@swift-ci Please test

File: tools/SourceKit/lib/SwiftLang/SwiftCompletion.cpp [hunk 1]
```diff
@@ -1236,6 +1236,7 @@ void SwiftLangSupport::codeCompleteOpen(
         typeContextKind = completionCtx.typeContextKind;
         mayUseImplicitMemberExpr = completionCtx.MayUseImplicitMemberExpr;
         consumer.setReusingASTContext(completionCtx.ReusingASTContext);
+        consumer.setAnnotatedTypename(completionCtx.getAnnnoateResult());
```
Comment (akyrtzi) [inline tools/SourceKit/lib/SwiftLang/SwiftCompletion.cpp:1239]: Typo `getAnnnoateResult`

File: lib/IDE/CodeCompletion.cpp [hunk 2]
```diff
@@ -3990,7 +4074,7 @@ class CompletionLookup final : public swift::VisibleDeclConsumer {
       builder.setLiteralKind(kind);
 
       consumer(builder);
-      addTypeRelationFromProtocol(builder, kind, defaultTypeName);
+      addTypeRelationFromProtocol(builder, kind);
```
Comment (benlangmuir) [inline lib/IDE/CodeCompletion.cpp:4077]: `defaultTypeName` is now unused, so you can remove it from the lambda and the callsites.  This applies to both the `addFromProto` lambdas in this patch.

**Review by benlangmuir — APPROVED** [review]

**Comment by rintaro** [comment]
@swift-ci Please smoke test

**Review by akyrtzi — APPROVED** [review]

**Comment by rintaro** [comment]
Windows: https://ci-external.swift.org/view/Pull%20Request/job/swift-PR-windows/2887

**Pull request merged**